File: programmers/greedy/jisu/구명보트.py
from collections import deque


def solution(people, limit):
    waiting = deque(sorted(people))
    boat = []
    num_boat = 0
    while waiting:
        sum_boat = 0
        while waiting and limit-sum_boat >= waiting[-1]:
            person = waiting.pop()
            boat.append(person)
            sum_boat += person
        while waiting and limit-sum_boat >= waiting[0]:
            person = waiting.popleft()
            boat.append(person)
            sum_boat += person
        num_boat += 1

    return num_boat

# 최대 두명만 탈 수 있다는 조건을 더하지 않았다
# 투포인터로 풀 수 있음

File: programmers/greedy/sangwoo/구명보트.js
function solution(people, limit) {
    var answer = 0;
    people = people.sort((a,b)=>b-a).map(v=>Number(v));
    
    for(let LN=0, RN = people.length -1;LN<=RN ; LN++){
        if(people[LN]+people[RN] <= limit) RN--;
        answer++;
    }
    console.log(answer);
    return answer;
}